cli/tests/pcluster/configure/test_pcluster_configure/test_no_available_no_input_no_automation_no_errors_with_config_file/pcluster.config.ini

```diff
@@ -10,6 +10,8 @@ compute_instance_type = t2.nano
 max_queue_size = 14
 initial_queue_size = 13
 maintain_initial_size = true
+base_os = alinux2
+additional_iam_policies = arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy
```

Comment:
Why is this policy, or any `additional_iam_policies`, expected to be added by `pcluster configure`?

Reply:
It's a bug with the current implementation. This the default policy that the config logic adds behind the scenes when cloudwatch logging is enabled. There's another one that's added when Batch is used as the scheduler.

Reply:
I'll add an entry to the backlog for this.

Reply:
I looked more into this, and it's a bug related to the handling of loading a config that uses one region, and then changing it.

In this case, the region starts as eu-west-1. Since CloudWatch logging is enabled, the internal value of the parameter is set to ['arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy'].

When the region is updated to cn-north-1, the internal value is updated to ['arn:aws-cn:iam::aws:policy/CloudWatchAgentServerPolicy', 'arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy'].

Then when the value is converted to a string and the dynamically added parameters are removed, only the policy specific to the currently configured partition is removed, and thus arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy is written back.

Reply:
Ok with leaving the fix for this out of this PR, but I think we should fix this ASAP.

Generally ok with the rest of this PR, waiting for another reviewer to give a second look.